### PR TITLE
fix(platform): fix fdp-table popover alignment

### PR DESCRIPTION
**Pull request by khotcholava**
closes [#11230](https://github.com/SAP/fundamental-ngx/issues/11230)

## Description
Ensure that the column header alignment is preserved when sorting columns in the fdp-table component. Previously, the header would jump to the start regardless of the initial alignment when sorting was activated. The header styles have been adjusted to maintain the specified alignment.

Additionally, the popover used for sorting is now aligned according to the column alignment, rather than always defaulting to the left.

### CODE REVIEW

File: libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts [hunk 1]
```diff
@@ -189,4 +191,18 @@ export class TableCellHeaderPopoverComponent implements AfterViewInit {
             this._listComponent?._setCurrentActiveItemIndex(0);
         }
     }
+
```
Comment (dpavlenishvili) [inline libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts:194]: create enum for placement strings

Comment (InnaAtanasova) [reply libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts:194]: there's enum `ColumnAlign` that can be imported and used

File: libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts [hunk 2]
```diff
@@ -189,4 +191,18 @@ export class TableCellHeaderPopoverComponent implements AfterViewInit {
...
+    /** @hidden */
+    mapColumnAlignToPlacement(columnAlignValue: ColumnAlignValue): Placement {
+        switch (columnAlignValue) {
+            case 'start':
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts:198]: case ColumnAlign.START

File: libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts [hunk 3]
```diff
@@ -189,4 +191,18 @@ export class TableCellHeaderPopoverComponent implements AfterViewInit {
...
+                return 'bottom-start';
+            case 'center':
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts:200]: case ColumnAlign.CENTER

File: libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts [hunk 4]
```diff
@@ -189,4 +191,18 @@ export class TableCellHeaderPopoverComponent implements AfterViewInit {
...
+                return 'bottom';
+            case 'end':
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.ts:202]: case ColumnAlign.END

**Review by InnaAtanasova — APPROVED** [review]

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
I set `[sortable]="true"` on this price column in the docs playground and it's still left aligned:
Screenshot 2024-09-03 at 3 35 45 PM

**Review by mikerodonnell89 — APPROVED** [review]
Actually appears to be working fine, maybe I had a cache issue or was looking at the wrong branch locally. approving 👍

**Review by dpavlenishvili — APPROVED** [review]

**Pull request merged**